apps/web/components/I18nLanguageHandler.tsx

```diff
@@ -1,27 +1,33 @@
+import { useSession } from "next-auth/react";
 import { useTranslation } from "next-i18next";
 import { useEffect } from "react";
 
 import { trpc } from "@calcom/trpc/react";
 
-export function useViewerI18n() {
-  return trpc.viewer.public.i18n.useQuery(undefined, {
-    staleTime: Infinity,
-    /**
-     * i18n should never be clubbed with other queries, so that it's caching can be managed independently.
-     * We intend to not cache i18n query
-     **/
-    trpc: {
-      context: { skipBatch: true },
-    },
-  });
+// eslint-disable-next-line turbo/no-undeclared-env-vars
+const vercelCommitHash = process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA || "NA";
```

Comment:
The commit hash should be enough for now. WDYT @keithwillcode

Reply:
Yeah that works too!

Reply:
@zomars thinking about this more; if someone is self-hosting and not on Vercel, this would request the i18n endpoint with NA, which means the next time they upgrade, the new translations wouldn’t be pulled by the browser because the request URL would be the same.

Reply:
I was having this exact problem, when running locally when I changed the translation files everything would work fine. But when deployed to remote the translation would always be the old ones. Using an env var of commit hash is far from being a good approach. Especially if it is not written anywhere about it.

Reply:
I believe I'm having an issue related with this, when I deploy the changes to production the translations are never updated. Could it be the reason?